### PR TITLE
feat: add foundry to CI 

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

Forge tests should run in CI. this PR does this by modifying the core `package.json` file.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -74,7 +74,8 @@
   ],
   "sideEffects": false,
   "scripts": {
-    "test": "yarn hardhat-test",
+    "test": "yarn hardhat-test && yarn forge-test",
+    "forge-test":"./scripts/SetUpFoundryWithHardhat.sh && forge test",
```
Comment (chrismaree) [inline packages/core/package.json:78]: note that this is a no-op if foundry is already setup and installed.

**Review by md0x — APPROVED** [review]
LGTM!

File: .circleci/config.yml [hunk 2]
```diff
@@ -15,6 +15,12 @@ jobs:
       - run:
           name: Install Dependencies
           command: yarn --frozen-lockfile
+      - run:
+          name: Install Foundry
+          command: |
+            curl -L https://foundry.paradigm.xyz | bash || true
```
Comment (mrice32) [inline .circleci/config.yml:21]: Why the `|| true`? Does that mean we ignore failures?

File: packages/core/package.json (hunk 1 above)
Comment (mrice32) [reply packages/core/package.json:78]: Why do we use this script here, but `curl -L https://foundry.paradigm.xyz` above?

File: .circleci/config.yml (hunk 2 above)
Comment (chrismaree) [reply .circleci/config.yml:21]: yup, that's correct! reason is that this installs installs Foundry but it's path is broken after the install due to this being in circleCI. the `|| true` means that we ignore the error that is produced but continue with the execution. Later, we configure the environment path to correctly make foundry accessible.  it's a bit janky...I'll admit... but it works.

**Comment by chrismaree** [comment]
https://github.com/UMAprotocol/protocol/pull/4247#discussion_r1029721111

cant directly reply to your comment on this one for some reason but this script does a bunch more than just install it. it sets up the repository to work with foundry by configuring the foundry standard lib within protocol that is needed for the execution.

**Pull request merged**